theme-kit/intl.tsx

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import 'server-only';
+
+import { getSupportedLocaleIsoCode } from '@prezly/theme-kit-core';
+import type { Locale } from '@prezly/theme-kit-intl';
+import type { ReactElement } from 'react';
+
+import { locale } from './locale';
+
+type Descriptor = {
+    id: string;
+    defaultMessage?: string;
+};
+
+type Values = Record<string, string | ReactElement>;
+
+interface Props {
+    values?: Values;
+    locale?: Locale.Code;
+    for: Descriptor;
+}
+
+export async function FormattedMessage(props: Props) {
+    const { values } = props;
+    const { id, defaultMessage } = props.for;
+    const { formatMessage } = await i18n(props.locale);
+
+    return formatMessage({ id, defaultMessage }, values);
+}
+
+export async function i18n(code?: Locale.Code) {
+    const dictionary = await importDictionary(code ?? locale());
+
+    return {
+        formatMessage(descriptor: Descriptor, values: Values = {}) {
+            return formatIntlMessage(dictionary, descriptor, values);
+        },
+    };
+}
+
+type Dictionary = Record<string, MessageFormat>;
+type MessageFormat = { type: 0 | 1; value: string }[];
+
+async function importDictionary(code: Locale.Code): Promise<Dictionary> {
+    const fileCode = getSupportedLocaleIsoCode(code);
+
+    return import(`@prezly/theme-kit-intl/messages/${fileCode}.json`);
+}
+
+function formatIntlMessage(
```

Comment:
`react-intl` also has `formatMessage` function returned from `useIntl` hook, that is used for places where JSX is not supported (e.g. `title` property on a link). 
Would it be possible to also implement that here? Then we'll likely be able to drop the `react-intl` dependency completely.

Reply:
> Then we'll likely be able to drop the react-intl dependency completely.

Yeah, I have been thinking about this too :) 

Btw, the server-components version of this function is 10 lines above your comment :) 

https://github.com/prezly/theme-nextjs-bea/blob/71e397636bb1e869d1e5813e0d7fa4c11d25464e/theme-kit/intl.tsx#L31-L39

Reply:
> Btw, the server-components version of this function is 10 lines above your comment :)

Yes, but from what I saw in the code, it still renders JSX, right? We also need a version that returns a plain string for the cases I described above

Reply:
> We also need a version that returns a plain string for the cases I described above

Ah, that's a good point. I thought that `formatMessage()` and `FormattedMessage` are doing the same thing. But now I see they don't. Thank! : )

Reply:
910d2243215bdc558f05b0d070bdb8c79ac759b3